notebooks/advanced/Time_Series_Forecasting_with_Unstructured_Data_and_Amazon_SageMaker_Neural_Topic_Model/3_Forecast.ipynb

```diff
@@ -0,0 +1,972 @@
+   "outputs": [],
+   "source": [
+    "outdir = './forecast-data'\n",
+    "if not os.path.exists(outdir):\n",
+    "    os.mkdir(outdir)\n",
+    "\n",
+    "target_df.to_csv(os.path.join(outdir, 'target_time_series.csv').replace(\"\\\\\",\"/\"), index=False)\n",
+    "related_df.to_csv(os.path.join(outdir, 'related_time_series.csv').replace(\"\\\\\",\"/\"), index = False)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "s3.upload_file(Filename=\"./forecast-data/target_time_series.csv\", Bucket=bucket, Key=\"{}/{}\".format(prefix, 'target_time_series.csv')\n",
+    ")\n",
+    "s3.upload_file(Filename=\"./forecast-data/related_time_series.csv\", Bucket=bucket, Key=\"{}/{}\".format(prefix, 'related_time_series.csv')\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Define the dataset schemas to ingest into Forecast"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Forecast has a number of predefined **Domains** which come with predefined schemas for data ingestion. Since we are interested in *web traffic*, we choose the WEB_TRAFFIC domain below.\n",
+    "\n",
+    "This provides a predefined schema and attribute types for the attributes we include in the target and related time series. For the WEB_TRAFFIC domain, there is no item metadata, only target and related time series data is allowed. \n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Define the schema for the target time series"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Set the dataset name to a new unique value. If it already exists, go to the Forecast console and delete any existing\n",
+    "# dataset ARNs and datasets.\n",
+    "\n",
+    "datasetName = 'webtraffic_forecast_NLP'\n",
+    "\n",
+    "schema ={\n",
+    "   \"Attributes\":[\n",
+    "      {\n",
+    "         \"AttributeName\":\"item_id\",\n",
+    "         \"AttributeType\":\"string\"\n",
+    "      },    \n",
+    "       {\n",
+    "         \"AttributeName\":\"timestamp\",\n",
+    "         \"AttributeType\":\"timestamp\"\n",
+    "      },\n",
+    "      {\n",
+    "         \"AttributeName\":\"value\",\n",
+    "         \"AttributeType\":\"float\"\n",
+    "      }      \n",
+    "   ]\n",
+    "}\n",
+    "\n",
+    "try:\n",
+    "    response = forecast.create_dataset(\n",
+    "                    Domain=\"WEB_TRAFFIC\",\n",
+    "                    DatasetType='TARGET_TIME_SERIES',\n",
+    "                    DatasetName=datasetName,\n",
+    "                    DataFrequency=DATASET_FREQUENCY, \n",
+    "                    Schema = schema\n",
+    "                   )\n",
+    "    datasetArn = response['DatasetArn']\n",
+    "    print('Success')\n",
+    "except Exception as e:\n",
```

Comment:
Catching generic "Exception" class to skip creating existing resources, but shouldn't do this as it is possible to catch different types of errors.
Use forecast.exceptions.ResourceAlreadyExistsException specifically.